consensus/rules/src/blocks.rs

```diff
@@ -122,10 +122,10 @@ pub fn check_block_pow(hash: &[u8; 32], difficulty: u128) -> Result<(), BlockErr
 /// Returns the penalty free zone
 ///
 /// <https://cuprate.github.io/monero-book/consensus_rules/blocks/weight_limit.html#penalty-free-zone>
-pub fn penalty_free_zone(hf: &HardFork) -> usize {
-    if hf == &HardFork::V1 {
+pub fn penalty_free_zone(hf: HardFork) -> usize {
+    if hf == HardFork::V1 {
```

Comment:
Probably >50% of the diffs in this PR are from changing `&HardFork` signatures to `HardFork`.

`HardFork` is 1 byte so it's cheaper to copy it rather than creating/using an 8 byte pointer to it.